examples/closure-app/silabs/src/ClosureManager.cpp

```diff
@@ -198,24 +207,78 @@ void ClosureManager::HandleClosureActionCompleteEvent(AppEvent * event)
             instance.HandleClosureActionComplete(instance.GetCurrentAction());
         });
         break;
+    case Action_t::PANEL_STEP_ACTION:
+        PlatformMgr().ScheduleWork([](intptr_t) {
+            ClosureManager & instance = ClosureManager::GetInstance();
+            instance.HandlePanelStepAction(instance.mCurrentActionEndpointId);
+        });
+        break;
+    default:
+        break;
+    }
+}
+
+void ClosureManager::HandleClosureActionComplete(Action_t action)
+{
+    ClosureManager & instance = ClosureManager::GetInstance();
+
+    switch (action)
+    {
+    case Action_t::CALIBRATE_ACTION: {
+        instance.ep1.OnCalibrateActionComplete();
+        instance.ep2.OnCalibrateActionComplete();
+        instance.ep3.OnCalibrateActionComplete();
+        instance.isCalibrationInProgress = false;
+        break;
+    }
+    case Action_t::STOP_MOTION_ACTION:
+        // This should handle the completion of a stop motion action.
+        break;
+    case Action_t::STOP_CALIBRATE_ACTION:
+        // This should handle the completion of a stop calibration action.
+        break;
+    case Action_t::MOVE_TO_ACTION:
+        // This should handle the completion of a move to action.
+        break;
+    case Action_t::PANEL_STEP_ACTION:
+        instance.ep1.OnPanelMotionActionComplete();
+        if (instance.mCurrentActionEndpointId == instance.ep2.GetEndpoint())
+        {
+            instance.ep2.OnPanelMotionActionComplete();
+        }
+        else if (instance.mCurrentActionEndpointId == instance.ep3.GetEndpoint())
+        {
+            instance.ep3.OnPanelMotionActionComplete();
+        }
+        instance.isStepActionInProgress = false;
+        break;
     default:
+        ChipLogError(AppServer, "Invalid action received in HandleClosureAction");
         break;
     }
+    // Reset the current action after handling the closure action
+    instance.SetCurrentAction(Action_t::INVALID_ACTION);
+    instance.mCurrentActionEndpointId = kInvalidEndpointId;
 }
 
 chip::Protocols::InteractionModel::Status ClosureManager::OnCalibrateCommand()
 {
     VerifyOrReturnValue(ep1.GetLogic().SetCountdownTimeFromDelegate(kCountdownTimeSeconds) == CHIP_NO_ERROR, Status::Failure,
                         ChipLogError(AppServer, "Failed to set countdown time for calibration"));
 
+    SetCurrentAction(CALIBRATE_ACTION);
+    mCurrentActionEndpointId = ep1.GetEndpoint();
+
+    isCalibrationInProgress = true;
```

Comment:
Should we check that there isn't already a Calibration in progress before setting all those members? Or is it ok to overwrite the calibration in progress?

Since most of the actions are dealt asynchronously, it seems there is a lack of thread safetyness with the implementation.